core/src/main/scala/org/http4s/Message.scala

```diff
@@ -447,6 +447,7 @@ object Request {
     val PathTranslated: AttributeKey[File] = AttributeKey[File]
     val ConnectionInfo: AttributeKey[Connection] = AttributeKey[Connection]
     val ServerSoftware: AttributeKey[ServerSoftware] = AttributeKey[ServerSoftware]
+    val SecureSession: AttributeKey[Option[SecureSession]] = AttributeKey[Option[SecureSession]]
```

Comment:
What does it mean if the key is present, but `None`?  That it's supported by the backend, but that the session wasn't secure?

Reply:
Yes.  It wasn't clear on the preferred convention to express that secure session values are optional.  For guidance, I saw `ServletRequestKeys.HttpSession(Option(req.getSession(false)))` in Http4sServlet.scala, so I followed that lead and went with a required key and optional value.  Also the name `SecureSession` here is meant to correspond the 'secure' in the `isSecure` method of the request.

Reply:
Okay, yes, I agree.